extension/ballastextension/go.mod

```diff
@@ -53,6 +53,8 @@ replace go.opentelemetry.io/collector/exporter => ../../exporter
 
 replace go.opentelemetry.io/collector/extension => ../
 
+replace go.opentelemetry.io/collector/extension/auth => ../auth
```

Comment:
Interesting, why does the ballast extension has a dependency on the auth package?

Reply:
it does not... crosslink adds the replace in here automatically

Reply:
Is there any chance that this is a transient dependency?

Reply:
I removed the replace statement, ran the tests, and it worked without implying further changes to go.sum nor go.mod. Meaning: it's not a transitive dependency.